src/orquestra/sdk/_base/_spaces/_resolver.py

```diff
@@ -0,0 +1,39 @@
+################################################################################
+# © Copyright 2023 Zapata Computing Inc.
+################################################################################
+import os
+from typing import Optional
+
+from orquestra.sdk.exceptions import ProjectInvalidError
+from orquestra.sdk.schema.workflow_run import ProjectId, WorkspaceId
+
+from .._config import AUTO_CONFIG_NAME
+from .._env import CURRENT_PROJECT_ENV, CURRENT_WORKSPACE_ENV
+from ._structs import ProjectRef
+
+
+def resolve_studio_project_ref(
+    workspace_id: Optional[WorkspaceId],
+    project_id: Optional[ProjectId],
+    config_name: str,
+) -> Optional[ProjectRef]:
+    # Passed explicitly
+    if workspace_id or project_id:
+        if workspace_id is None or project_id is None:
+            raise ProjectInvalidError(
+                "Invalid project ID. Either explicitly pass workspace_id "
+                "and project_id, or omit both"
+            )
+        return ProjectRef(workspace_id=workspace_id, project_id=project_id)
```

Comment:
Alternative suggestion:
```suggestion
if workspace_id and project_id:
    return ProjectRef(workspace_id=workspace_id, project_id=project_id)
elif workspace_id or project_id:
    raise ProjectInvalidError(
        "Invalid project ID. Either explicitly pass workspace_id "
        "and project_id, or omit both"
    )
```